src/main/java/org/apache/sysml/hops/rewrite/HopRewriteUtils.java

```diff
@@ -241,7 +241,24 @@ public static void addChildReference( Hop parent, Hop child, int pos ){
 		parent.getInput().add( pos, child );
 		child.getParent().add( parent );
 	}
-	
+
+	/**
+	 * Replace an old Hop with a replacement Hop.
+	 * If the old Hop has no parents, then return the replacement.
+	 * Otherwise rewire each of the Hop's parents into the replacement and return the replacement.
+	 * @param old To be replaced
+	 * @param replacement The replacement
+	 * @return replacement
+	 */
+	public static Hop replaceHop(final Hop old, final Hop replacement) {
+		final ArrayList<Hop> rootParents = old.getParent();
+		if (rootParents.isEmpty())
```

Comment:
This seems like a generally useful method. It correctly rewires an old Hop to a new one regardless of whether the old Hop has parents (i.e. is interior) or not (i.e. is a root). 

Though your logic is sound, why make developers think hard about whether they can safely assume a Hop could be a root or not when they could use one method that always works? Also, if dead code elimination didn't run for some reason, then the code without this method would become buggy, so that option seems less resilient.

It's a minor thing, but I think using a method like this to do rewiring would make life easier for developers.